python/MyriaPythonWorker.py

```diff
@@ -123,6 +124,9 @@ def write_with_length(cls, obj, stream, output_type):
         elif output_type == DataType.BLOB:
             cls.write_int(DataType.BLOB, stream)
             cls.pickle_and_write(obj, stream)
+        elif output_type ==DataType.EXCEPTION:
+            cls.write_int(len(obj), stream)
```

Comment:
If the object is always a `str`, then there is no reason to convert it with `str()`. You should just write the bytes directly. If you need to ensure that it's always a `str`, then add an assert `type(obj) is str`. If the object could potentially be another type, then we should always explicitly call `unicode(obj).encode("utf-8")` in the method.